common/src/main/resources/META-INF/openapi.json

```diff
@@ -2786,6 +2786,130 @@
                 }
             ]
         },
+        "/groups": {
+            "summary": "Collection the groups in Registry",
+            "get": {
+                "tags": [
+                    "Groups"
+                ],
+                "responses": {
+                    "200": {
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "type": "array",
+                                    "items": {
+                                        "type": "string"
+                                    }
```

Comment:
I think this operation should return an array of `GroupMetaData` instead.  I think returning an array of strings is a bit of an anti-pattern.  The only reason not to return the metadata for the group is if we think it takes a long time to fetch.

Reply:
Actually it's more than this.  I think this needs to return a paged collection of `GroupMetaData`.  So we need a new `GroupSearchResults` type I think.  See the `GET` operation for the `/groups/{groupId}/artifacts` endpoint.  I think we should follow that pattern.

Reply:
I tend to agree with this and is one of the doubts I had when I created this operation but I wanted to be consistent with the CNCF API and this is what that API is doing,